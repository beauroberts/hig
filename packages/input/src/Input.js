import React from "react";

import inputPropTypes from "./inputPropTypes";
import InputBehavior from "./behaviors/InputBehavior";
import InputPresenter from "./presenters/InputPresenter";

function Input({ value, disabled, onInput, onChange, onFocus, onBlur }) {
  return (
    <InputBehavior onBlur={onBlur} onFocus={onFocus}>
      {({
        handleBlur,
        handleFocus,
        hasFocus,
        hasHover,
        handleMouseEnter,
        handleMouseLeave
      }) => (
        <InputPresenter
          value={value}
          disabled={disabled}
          hasHover={hasHover}
          hasFocus={hasFocus}
          onBlur={handleBlur}
          onChange={onChange}
          onFocus={handleFocus}
          onInput={onInput}
          onMouseEnter={handleMouseEnter}
          onMouseLeave={handleMouseLeave}
        />
      )}
    </InputBehavior>
  );
}

Input.propTypes = inputPropTypes;

export default Input;
