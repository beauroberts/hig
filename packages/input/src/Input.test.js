import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import sinon from "sinon";

import Input from "./index";

function snapshotTest(props) {
  const tree = renderer.create(<Input {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
}

describe("Input", () => {
  it("renders children", () => {
    snapshotTest({
      value: "jon.snow@winterfell.com",
      onChange: () => {},
      onFocus: () => {},
      onBlur: () => {},
      onInput: () => {}
    });
  });

  describe("events", () => {
    it("on focus calls the onFocus callback", () => {
      const spy = sinon.spy();
      const wrapper = mount(<Input onFocus={spy} />);

      wrapper
        .find("input")
        .first()
        .simulate("click");

      expect(spy.calledOnce).toBe(true);
    });

    it("on blur calls the onBlur callback", () => {
      const spy = sinon.spy();
      const wrapper = mount(<Input onBlur={spy} />);

      wrapper
        .find("input")
        .first()
        .simulate("click")
        .simulate("blur");

      expect(spy.calledOnce).toBe(true);
    });

    it("on keypress calls the onInput callback", () => {
      const spy = sinon.spy();
      const wrapper = mount(<Input onInput={spy} />);

      wrapper
        .find("input")
        .first()
        .simulate("keypress", { key: "f" });

      expect(spy.calledOnce).toBe(true);
    });

    it("on change calls the onChange callback", () => {
      const spy = sinon.spy();
      const wrapper = mount(<Input onChange={spy} />);

      wrapper
        .find("input")
        .first()
        .simulate("keypress", { key: "f" })
        .simulate("blur");

      expect(spy.calledOnce).toBe(true);
    });
  });
});
