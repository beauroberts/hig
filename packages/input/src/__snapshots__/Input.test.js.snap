// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders children 1`] = `
<div
  className="css-g8jj0w"
>
  <input
    className="css-da639k"
    disabled={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    value="jon.snow@winterfell.com"
  />
  <div
    className="css-1u12m60"
  />
</div>
`;
